### DISCUSSION TITLE
Copilot Intellij - unable to verify the first certificate

### DISCUSSION THREAD

**Discussion by adamgrze-luma** [community/community#120205] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

GitHUb Copilot: 1.5.2.5345
IntelliJ: IntelliJ IDEA 2024.1 (Ultimate Edition) Build #IU-241.14494.240, built on March 28, 2024
OS: Windows 10

Right after starting Intellij, copilot only shows one suggestion and then crashes with "unable to verify the first certificate".
![image](https://github.com/community/community/assets/103991899/48557715-7c22-4f29-8cc8-7c3c8a660778)

**Logs:**
2024-04-18 19:09:02,248 [3921084]   WARN - #copilot - [ghostText] Error on ghost text request: FetchError: The pending stream has been canceled (caused by: unable to verify the first certificate)
    at fetch (C:\snapshot\copilot-client\node_modules\@adobe\helix-fetch\src\fetch\index.js:99:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at cachingFetch (C:\snapshot\copilot-client\node_modules\@adobe\helix-fetch\src\fetch\index.js:288:16)
    at WJ.fetch (C:\snapshot\copilot-client\lib\src\network\helix.ts:78:22)
    at oH.fetchWithParameters (C:\snapshot\copilot-client\lib\src\openai\fetch.ts:426:26)
    at oH.fetchAndStreamCompletions (C:\snapshot\copilot-client\lib\src\openai\fetch.ts:321:26)
    at genericGetCompletionsFromNetwork (C:\snapshot\copilot-client\lib\src\ghostText\ghostText.ts:143:21)
    at getGhostText (C:\snapshot\copilot-client\lib\src\ghostText\ghostText.ts:692:23)
    at getGhostTextWithAbortHandling (C:\snapshot\copilot-client\agent\src\methods\getCompletions.ts:302:16)
    at handleGetCompletionsHelper (C:\snapshot\copilot-client\agent\src\methods\getCompletions.ts:154:33) {
  type: 'system',
  _name: 'FetchError',
  code: 'ERR_HTTP2_STREAM_CANCEL',
  errno: undefined,
  erroredSysCall: undefined
}

The same plugin in Visual Studio Code works fine.

**I've already tried:**
1) Log out and log back in to GitHub
2) Reinstall the plugin
3) Use 3 previous versions of the plugin
4) Run the plugin on a fresh Intellij installation

**Comment by michieldondorp** [comment, marked as answer]
This was also happening on our end, it appeared to be eset antivirus security setting (settings -> advanced -> security -> ssl/tls)

**Reply by maxwell-01** [reply]
Thanks for the reply, I have this error and ESET installed. 
What did you change? Is that ssl/tls a menu in the IDE or in eset?

**Reply by adamgrze-luma** [reply]
@michieldondorp you are genius! Thanks! I added "Ignore" action for "copilot-agent-win.exe" and it seems to be working now 👍 

![image](https://github.com/community/community/assets/103991899/803eaecd-34aa-42fe-8189-ffb2f294e71a)

What's interesting is why it worked fine for months and then suddenly stopped only for IntelliJ. The plugin requires any changes?

**Reply by maxwell-01** [reply]
Thanks for sharing! I've contacted my internal IT team to see if they can have a go at this on my machine.

**Reply by TheGenezis** [reply]
@michieldondorp you are a lifesaver, I was going crazy as it just kept disconnecting me periodically, and it was impossible to google out the issue yesterday. Also, it seems like this "conflict" might been introduced only very recently as I was using the same setup for a long time without this particular issue

**Reply by Johnparke888** [reply]
This worked for me also. I have been using copilot for months, then suddenly I start getting error messages. I suspected eset, but didn't know about the advanced settings. Thanks, John

**Reply by petrsol** [reply]
This worked for me too. Thanks! Using Preview version of MSVC, my copilot-agent-win.exe is on different path: C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\Extensions\VisualStudio.GitHub.Copilot\service\dist\copilot-agent-win.exe.
Also using ESET INTERNET SECURITY and settings path is different: Setup / Advanced Setup / Protections / SSL/TLS / Aplication scan rules - Edit.

**Reply by BewaAutomatisierung-RD** [reply]
Does anyone know the path of copilot for Visual Studio 2022.  I get the same error there but can't find an .exe for the github copilot extension. The path mentioned by @petrsol does not exist on my machine (even without the "Preview").

**Reply by BewaAutomatisierung-RD** [reply]
Found it. It's located here: 
"C:\Users\<username>\AppData\Local\Microsoft\VisualStudio\17.0_ec60bf4b\Extensions\okrqpxul.pny\service\dist\copilot-agent-win.exe"

**Reply by Matt-PMCT** [reply]
This solution shouldn't be considered the permanent fix; we are putting an ESET exception in on a file that is located in essentially a temporary directory. 

Every time the IDE updates to a new version, we will have to add a new exception. The root issue seems to be something that Copilot introduced recently since none of us had this problem before. That root issue should be addressed IMO.

**Reply by JPG-Digital** [reply]
Thank you!

**Reply by SilentHG** [reply]
Facing similar issue, I am on a company laptop so modifying ESET internals are cumbersome process, please fix github

**Comment by Jaben** [comment]
Yes -- this is actually a deal-breaker for my company. I cannot enable this for my guys unless this is fixed.

**Comment by ohalushchak** [comment]
I have the same issue.

Here is error message.

**Sign in failed. Reason: Request signInInitiate failed with message: The pending stream has been canceled (caused by: unable to verify the first certificate), request id: 155, error code: -32603**

GitHub please fix it.

**Reply by dschuck** [reply]
Did you see the answer from @michieldondorp? This worked for me. If you / your company is using ESET Protection, you could try to add the "copilot-agent-win.exe" to the list of SSL/TLS filtered applications - the screenshot from @adamgrze-luma shows perfectly what to do in your ESET client.

**Reply by ohalushchak** [reply]
Ahhh. Sorry I did not notice this answer. Yes, this fix works for me, but any way it's more like workaround then good solution.

**Comment by otami** [comment]
In my machine, I have confirmed that this problem can also be solved by additionally setting ESET's own certificate for NPM.

In this case, no ESET configuration was required.

The following link has details on how to do it.
https://github.com/the-last-byte/ESET-NPM-Breakage-Fix/issues/1#issuecomment-2075166058

**Comment by thomaspemisia** [comment]
I had the same issue on a Windows 11 machine, VS 2022.
I confirm that the ESET SSL rule corrected the problem.
Thanks, everyone!

![image](https://github.com/community/community/assets/159035650/55286c10-3fb7-410d-8abb-40a56823ef36)
